crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -739,12 +751,68 @@ pub fn split(lhs: &InternalVar, bit_size: u32, evaluator: &mut Evaluator) -> Vec
     result
 }
 
+fn const_and(
+    var: InternalVar,
+    b: FieldElement,
+    bit_size: u32,
+    evaluator: &mut Evaluator,
+) -> Arithmetic {
+    let a_bits = split(&var, bit_size, evaluator);
+    let mut result = Arithmetic::default();
+    let mut k = FieldElement::one();
+    let two = FieldElement::from(2_i128);
+    for (a_iter, b_iter) in a_bits.into_iter().zip(b.bits().iter().rev()) {
+        if *b_iter {
+            result = add(&result, k, &from_witness(a_iter));
+        }
+        k = k.mul(two);
```

Comment:
Wouldn't looping over every bit and adding it individually be rather slow at runtime? How would this be faster than a non-const and?

Reply:
We have to do this because we do not have a custom gate for const and.

Reply:
How would this be faster than a non-const and? All I see is we create a Gate::And in the non-const case.

Reply:
My point is that the non-const and does not support const arguments, so it just don't work. Comparing speed makes no sense here.

Reply:
I don't know, but this is something we could look for in future.